drivers/flash/spi_flash_at45.c

```diff
@@ -549,23 +549,28 @@ static int spi_flash_at45_init(const struct device *dev)
 
 #if ANY_INST_HAS_RESET_GPIOS
 	if (dev_config->reset) {
-		if (gpio_pin_configure_dt(dev_config->reset,
-					GPIO_OUTPUT_ACTIVE)) {
+		if (!device_is_ready(dev_config->reset->port)) {
+			LOG_ERR("Reset pin not ready");
+			return -ENODEV;
+		}
+		if (gpio_pin_configure_dt(dev_config->reset, GPIO_OUTPUT_ACTIVE)) {
 			LOG_ERR("Couldn't configure reset pin");
 			return -ENODEV;
 		}
-		gpio_pin_set(dev_config->reset->port, dev_config->reset->pin, 0);
+		gpio_pin_set_dt(dev_config->reset, 0);
```

Comment:
No point setting it up as active to them make it inactive, can just use `GPIO_OUTPUT_INACTIVE` instead

Reply:
> No point setting it up as active to them make it inactive, can just use `GPIO_OUTPUT_INACTIVE` instead

It has an internal reset controller, we need to assert to trigger.